src/lib/operators/jit_operator/operators/jit_aggregate.cpp

```diff
@@ -0,0 +1,312 @@
+    std::vector<double>& avg_values) {
+  Fail("Invalid aggregate");
+}
+
+void JitAggregate::after_query(Table& out_table, JitRuntimeContext& context) const {
+  auto num_columns = _aggregate_columns.size() + _groupby_columns.size();
+  ChunkColumns chunk_columns(num_columns);
+
+  // Add each groupby column to the output table
+  for (const auto& column : _groupby_columns) {
+    const auto data_type = column.hashmap_value.data_type();
+
+    resolve_data_type(data_type, [&](auto type) {
+      using ColumnDataType = typename decltype(type)::type;
+      // Get the std::vector containing the raw values (and conditionally, also get the is_null values).
+      // We then create a ValueColumn of the appropriate data type from these values. Since value columns use a
+      // pmr_concurrent_vector internally, this operation copies all values to a new vector of this type.
+      // However, using this type of vector within the operator in the first place is also suboptimal, since the
+      // prm_concurrent_vector performs a lot of synchronization. It is thus better to use the faster std::vector and
+      // perform a single copy in the end.
+      auto& values = context.hashmap.values[column.hashmap_value.column_index()].template get_vector<ColumnDataType>();
+      if (column.hashmap_value.is_nullable()) {
+        auto& is_null_values = context.hashmap.values[column.hashmap_value.column_index()].get_is_null_vector();
+        chunk_columns[column.table_position] = std::make_shared<ValueColumn<ColumnDataType>>(values, is_null_values);
+      } else {
+        chunk_columns[column.table_position] = std::make_shared<ValueColumn<ColumnDataType>>(values);
+      }
+    });
+  }
+
+  for (const auto& column : _aggregate_columns) {
+    const auto data_type = column.hashmap_value.data_type();
+
+    if (column.function == AggregateFunction::Avg) {
+      // Perform the post-processing for average aggregates.
+      resolve_data_type(data_type, [&](auto type) {
+        using ColumnDataType = typename decltype(type)::type;
+
+        // First get the vectors for both internal aggregates (SUM and COUNT).
+        DebugAssert(column.hashmap_value_2, "Invalid avg aggregate column.");
+        auto sum_column_index = column.hashmap_value.column_index();
+        auto& sum_values = context.hashmap.values[sum_column_index].template get_vector<ColumnDataType>();
+        auto count_column_index = column.hashmap_value_2.value().column_index();
+        auto& count_values = context.hashmap.values[count_column_index].get_vector<int64_t>();
+
+        // Then compute the averages.
+        auto avg_values = std::vector<double>(sum_values.size());
+        _compute_averages(sum_values, count_values, avg_values);
+
+        if (column.hashmap_value.is_nullable()) {
+          auto& is_null_values = context.hashmap.values[column.hashmap_value.column_index()].get_is_null_vector();
```

Comment:
`has_null_values`?

Reply:
I agree that the name is weird, but `has_null_values` is semantically incorrect. This is the values from the `is_null` vector (so the type is `std::vector<bool>`, not `bool`). I changed it to simply `null_values`.